src/formatter/simple.rs

```diff
@@ -1,361 +1,433 @@
-                    let failure_name = self.name_stack.join(" | ");
-                    self.failures.push(failure_name);
+    fn exit_suite(&mut self, report: &SuiteReport) {
+        let _ = writeln!(self.buffer, "\nfailures:");
+        let failed = mem::replace(&mut self.failed, vec![]);
+        for scope_stack in failed {
+            for (indent, scope) in scope_stack.into_iter().enumerate() {
+                match scope {
+                    ScopeInfo::Suite(info) => {
+                        let padding = Self::padding(indent);
+                        let label: &str = info.label.into();
+                        let _ = writeln!(self.buffer, "{}{} {:?}:", padding, label, info.name);
+                    }
+                    ScopeInfo::Context(info) => {
+                        let padding = Self::padding(indent);
+                        let label: &str = info.label.into();
+                        let _ = writeln!(self.buffer, "{}{} {:?}:", padding, label, info.name);
+                    }
+                    ScopeInfo::Example(info) => {
+                        let padding = Self::padding(indent);
+                        let label: &str = info.label.into();
+                        if let Some(failure) = info.failure {
+                            let _ = writeln!(self.buffer, "{}{} {:?}:", padding, label, info.name);
+                            let padding = Self::padding(indent + 1);
+                            let _ = writeln!(self.buffer, "{}{}", padding, failure);
+                        } else {
+                            let _ = writeln!(self.buffer, "{}{} {:?}", padding, label, info.name);
+                        }
+                    }
                 }
-                self.name_stack.pop();
-
-                let chr = if result.is_ok() {
-                    "."
-                } else {
-                    "F"
-                };
-                write!(self.buf, "{}", chr)
-            }
-            Event::FinishedRunner(result) => {
-                let failures_sum = self.failures_summary();
-                let summary = self.write_summary(result);
-                writeln!(self.buf,
-                         "\n\nFailed examples:\n{}{}",
-                         failures_sum,
-                         summary)
-            }
-            Event::EndDescribe => {
-                self.name_stack.pop();
-                Ok(())
             }
-            // _ => Ok(()),
+        }
+
+        let label = if report.failed == 0 {
+            "ok".green()
+        } else {
+            "FAILED".red()
         };
+        let _ = write!(self.buffer, "\ntest result: {}.", label);
+        let _ = write!(self.buffer, " {} passed", report.passed);
+        let _ = write!(self.buffer, "; {} failed", report.failed);
+        let _ = write!(self.buffer, "; {} ignored", report.ignored);
+        let _ = write!(self.buffer, "; {} measured", report.measured);
+        let _ = writeln!(self.buffer, "");
```

Comment:
Sure, but is this …

```rust
let _ = writeln!(self.buffer, "\ntest result: {}. {} passed; {} failed; {} ignored; {} measured", label, report.passed, report.failed, report.ignored, report.measured);
```

… really any better than this …

```rust
let _ = write!(self.buffer, "\ntest result: {}.", label);
let _ = write!(self.buffer, " {} passed", report.passed);
let _ = write!(self.buffer, "; {} failed", report.failed);
let _ = write!(self.buffer, "; {} ignored", report.ignored);
let _ = write!(self.buffer, "; {} measured", report.measured);
let _ = writeln!(self.buffer, "");
```

?

The latter neatly fits in 80 chars, neatly associates each value with its format location, can easily be extended for additional fields, …

And I'm not sure performance should be of higher concern than usability here, if that's what you mean.